hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/Util.java

```diff
@@ -119,6 +119,32 @@ public static String getSendingFacility(String auth) {
 		return clientId;
 	}
 	
+	
+    /**
+     * The purpose of this method is to convert messageControlId to hexadecimal string for MQ processing.
+     * Checks for messageControlId. If missing populates currentdatetime.
+     * Appends "0" to match replyQ corelationId(The corelationId must be string of 48 hexadecimal digit)
+     * @param str
+     * @return
+     */
+    public static String convertStringToHex(String str) {
+    	//JMB validates msgControlId and returns error response for mandatory field.
+    	//CorrelationId must be set to some value in order to avoid 'MQSeries failure"
+    	if(StringUtils.isBlank(str)) {
+    		str=getDateTime();
+    	}
+        StringBuffer hex = new StringBuffer();
```

Comment:
StringBuilder is preferable to StringBuffer unless you need thread safety...which I don't believe to be the case here.